### ISSUE TITLE
CLI Update

### ISSUE THREAD

**Issue by cdrage** [containers/podman-desktop#3708]
### Is your enhancement related to a problem? Please describe

As per https://github.com/containers/podman-desktop/issues/2523:

We should be able to see that there is an update available for the compose CLI.

### Describe the solution you'd like

Implement the "update" part of the UI

~**August 31st update:** Awaiting finalized UI/UX to show update mechanism~

Split into two issues:

- [x] https://github.com/containers/podman-desktop/issues/4689
- [x] https://github.com/containers/podman-desktop/issues/4690
- [x] #5035

**Comment by cdrage** [comment]
Unpostponed as discussions have lead to having the update information located in a separate settings page:

#3781

**Comment by cdrage** [comment]
Waiting for https://github.com/containers/podman-desktop/pull/4197 to be merged in and then implementing this.

**Comment by mairin** [comment]
![Settings _ CLI Tools Update Flow](https://github.com/containers/podman-desktop/assets/799683/4c3cd441-833f-412d-ad25-28aa5dd1b176)

Here are the update flow mockups!

**Comment by cdrage** [comment]
> ![Settings _ CLI Tools Update Flow](https://user-images.githubusercontent.com/799683/276618088-4c3cd441-833f-412d-ad25-28aa5dd1b176.png)
> 
> Here are the update flow mockups!

Thank you! I'll unblock this issue, I will however keep this postponed on the sprint planning and work on https://github.com/containers/podman-desktop/issues/3707 for now. I'll most likely look at the code that @axel7083 has contributed to help integrate the update functionality of this.

**Comment by cdrage** [comment]
When implementing, I'll make sure that the commits are authored by you @axel7083 if that's okay with you. I don't want to be copying-and-pasting your code as you are the original author. 

I'm unsure how much time you have recently, but feel free to reach out to me on Discord so we can chat!

**Comment by axel7083** [comment]
> When implementing, I'll make sure that the commits are authored by you @axel7083 if that's okay with you. I don't want to be copying-and-pasting your code as you are the original author.
> 
> I'm unsure how much time you have recently, but feel free to reach out to me on Discord so we can chat!

Thanks ! What's your discord ?

**Comment by cdrage** [comment]
> > When implementing, I'll make sure that the commits are authored by you @axel7083 if that's okay with you. I don't want to be copying-and-pasting your code as you are the original author.
> > I'm unsure how much time you have recently, but feel free to reach out to me on Discord so we can chat!
> 
> Thanks ! What's your discord ?

Unable to send you emails to your profile, but you should be able to find my name within the Podman discord channel #podman-desktop https://podman.io/community

**Comment by cdrage** [comment]
I've split up this issue into two separate issues and updated the initial message:
- [ ] https://github.com/containers/podman-desktop/issues/4689 (backend implementation)
- [ ] https://github.com/containers/podman-desktop/issues/4690 (front end implementation and logic)

The reasoning is that we already have https://github.com/containers/podman-desktop/pull/4092 as well as the fact that we will have to integrate some form of checking / logic for the front end in order to see what the current version is before offering the "update now!" button. 

We also need to allow usage through the API since extensions handle checking the version (ex. running the `docker-compose --version --short` and parsing periodically). This allows extensions to simply pass the github link / gitlab / http / etc and let PD handle the rest. I believe this was briefly talked about on a standup call.

Should we mark this issue as an epic @deboer-tim more of an epic since we are increasing the scope?
